src/utilities/kubectl-util.ts

```diff
@@ -19,17 +19,26 @@ function getExecutableExtension(): string {
     return '';
 }
 
+function getKubectlArch(): string {
```

Comment:
You need to add a check to validate if `kubectl` can be installed on that `arch` as well.

Reply:
The best way to check if `kubectl` can be installed on particular `arch` is to send a request to the downloadURL and see what is the response. This is because for some versions of `kubectl` the supported arch's are different, also there may darwin arm come in the future.

At the same time the request may fail because the version of `kubectl` is not valid (e.g. 1.30.0 does not exist). Or network error occurred. 

So what I can propose is to check if the exception is an instance of `HTTPError` in downloadKubectl [which is thrown in tool-cache lib](https://github.com/actions/toolkit/blob/master/packages/tool-cache/src/tool-cache.ts#L103) and in case when response is 404 throw an error with more detailed message (like 'The kubectl tool version `<ver>` for arch `<arch>` does not exist.')

What do you think about it?

Reply:
BTW the default message for HTTPError is `Unexpected HTTP response: ${httpStatusCode}` - https://github.com/actions/toolkit/blob/main/packages/tool-cache/src/tool-cache.ts#L20

Reply:
Fair enough. 

The `downloadKubectl` doesn't seem to float this error message up, to the logs. 
Could you amend it as well? So that if there's a `404` we can assume it's an arch related issue.

Reply:
Actually the error is propagated to the log. Here is an example - https://github.com/gtrubach/k8s-deploy-test/runs/2088601164?check_suite_focus=true#step:5:16 (or [raw log](https://pipelines.actions.githubusercontent.com/UAGCXeplpCZe8Boqwsqr09QgMHPfUA9cUS8kBJ3tP8dQqSI059/_apis/pipelines/1/runs/20/signedlogcontent/3?urlExpires=2021-03-11T18%3A47%3A45.1879667Z&urlSigningMethod=HMACV1&urlSignature=1amaEcIOh0vFXGwtrp8tUaaAO0IDj%2BfPzFUUs%2FaaIOg%3D)). With current code if there is a 404 error during kubectl download, the below red message will be shown.

![image](https://user-images.githubusercontent.com/10942688/110837796-c1ac0d80-82b2-11eb-8c42-5bbdaa316cfb.png)

The successful example of this action is here - https://github.com/gtrubach/k8s-deploy-test/runs/2088682804?check_suite_focus=true and here for arm linux - https://github.com/gtrubach/k8s-deploy-test/runs/1999747697

Reply:
Cool. This works! 😄